### PR TITLE
{bio}[foss/2016b] HTSeq 0.9.1

### PR DESCRIPTION
**Pull request by SethosII**
This adds the current version of `HTSeq` for the toolchain `foss-2016b`, mentioned in https://github.com/easybuilders/easybuild-easyconfigs/issues/5015.

The `sanity_check_command` of `HTSeq` doesn't fail, allthough the same command fails when the module is loaded afterwards because the `PYTHONPATH` is somewhat wrong? The error is:

```
htseq-count --help
Traceback (most recent call last):
  File "/easybuild/test/software/HTSeq/0.9.1-foss-2016b-Python-2.7.12/bin/htseq-count", line 4, in <module>
    __import__('pkg_resources').run_script('HTSeq==0.9.1', 'htseq-count')
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 719, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1504, in run_script
    exec(code, namespace, namespace)
  File "/easybuild/test/software/HTSeq/0.9.1-foss-2016b-Python-2.7.12/lib/python2.7/site-packages/HTSeq-0.9.1-py2.7-linux-x86_64.egg/EGG-INFO/scripts/htseq-count", line 3, in <module>
    import HTSeq.scripts.count
  File "/easybuild/test/software/HTSeq/0.9.1-foss-2016b-Python-2.7.12/lib/python2.7/site-packages/HTSeq-0.9.1-py2.7-linux-x86_64.egg/HTSeq/__init__.py", line 9, in <module>
    from _HTSeq import *
  File "__init__.pxd", line 155, in init HTSeq._HTSeq (src/_HTSeq.c:39805)
  File "/usr/lib/python2.7/dist-packages/numpy/__init__.py", line 180, in <module>
    from . import add_newdocs
  File "/usr/lib/python2.7/dist-packages/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File "/usr/lib/python2.7/dist-packages/numpy/lib/__init__.py", line 8, in <module>
    from .type_check import *
  File "/usr/lib/python2.7/dist-packages/numpy/lib/type_check.py", line 11, in <module>
    import numpy.core.numeric as _nx
  File "/usr/lib/python2.7/dist-packages/numpy/core/__init__.py", line 14, in <module>
    from . import multiarray
ImportError: cannot import name multiarray
```

So it tries to load the system `numpy` not the one from the module. When I try to import it manually (`python -c "import numpy"`) I get the same error. Reversing the path however works (`python -c "import sys; sys.path.reverse(); import numpy"`). The path looks as follows:

```
python -c "import sys; print sys.path"                                                                     
['', '/easybuild/test/software/HTSeq/0.9.1-foss-2016b-Python-2.7.12/lib/python2.7/site-packages/HTSeq-0.9.1-py2.7-linux-x86_64.egg', '/easybuild/test/software/Pysam/0.9.1.4-foss-2016b-Python-2.7.12/lib/python2.7/site-packages/pysam-0.9.1.4-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/matplotlib/1.5.3-foss-2016b-Python-2.7.12/lib/python2.7/site-packages/cycler-0.10.0-py2.7.egg', '/easybuild/16.04/software/matplotlib/1.5.3-foss-2016b-Python-2.7.12/lib/python2.7/site-packages/matplotlib-1.5.3-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/pyparsing-2.1.5-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/pytz-2016.4-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/python_dateutil-2.5.3-py2.7.egg', '/easybuild/test/software/Pysam/0.9.1.4-foss-2016b-Python-2.7.12/lib/python2.7/site-packages/Cython-0.25.1-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/EasyBuild/3.3.1/lib/python2.7/site-packages/vsc_install-0.10.26-py2.7.egg', '/usr/lib/python2.7/dist-packages', '/easybuild/16.04/software/EasyBuild/3.3.1/lib/python2.7/site-packages/vsc_base-2.5.8-py2.7.egg', '/easybuild/16.04/software/EasyBuild/3.3.1/lib/python2.7/site-packages/easybuild_framework-3.3.1-py2.7.egg', '/easybuild/16.04/software/EasyBuild/3.3.1/lib/python2.7/site-packages/easybuild_easyblocks-3.3.1-py2.7.egg', '/easybuild/16.04/software/EasyBuild/3.3.1/lib/python2.7/site-packages/easybuild_easyconfigs-3.3.1-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/setuptools-23.1.0-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/pip-8.1.2-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/numpy-1.11.1-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/blist-1.3.6-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/Cython-0.24-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/six-1.10.0-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/python_dateutil-2.5.3-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/decorator-4.0.10-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/liac_arff-2.1.0-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/ecdsa-0.13-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/cryptography-1.4-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/ipaddress-1.0.17-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/enum34-1.1.6-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/pyasn1-0.1.9-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/idna-2.2-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/paramiko-2.0.1-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/pyparsing-2.1.5-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/netifaces-0.10.4-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/funcsigs-1.0.2-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/pytz-2016.4-py2.7.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/pandas-0.18.1-py2.7-linux-x86_64.egg', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages/virtualenv-15.0.2-py2.7.egg', '/easybuild/test/software/HTSeq/0.9.1-foss-2016b-Python-2.7.12/lib/python2.7/site-packages', '/easybuild/16.04/software/matplotlib/1.5.3-foss-2016b-Python-2.7.12/lib/python2.7/site-packages', '/easybuild/test/software/Pysam/0.9.1.4-foss-2016b-Python-2.7.12/lib/python2.7/site-packages', '/easybuild/test/software/numpy/1.10.1-foss-2016b-Python-2.7.12/lib/python2.7/site-packages', '/easybuild/16.04/software/EasyBuild/3.3.1/lib/python2.7/site-packages', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python27.zip', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/plat-linux2', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/lib-tk', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/lib-old', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/lib-dynload', '/easybuild/16.04/software/Python/2.7.12-foss-2016b/lib/python2.7/site-packages']
```

### CODE REVIEW

File: easybuild/easyconfigs/h/HTSeq/HTSeq-0.9.1-foss-2016b-Python-2.7.12.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ('Python', '2.7.12'),
+    ('matplotlib', '1.5.3', versionsuffix),
+]
+
+sanity_check_paths = {
+    'files': ["bin/htseq-count", "bin/htseq-qa"],
+    'dirs': ["lib/python%(pyshortver)s/site-packages/%(name)s-%(version)s-py%(pyshortver)s-linux-x86_64.egg"],
+}
+
+sanity_check_commands = [('htseq-count', '--help')]
```
Comment (boegel) [inline easybuild/easyconfigs/h/HTSeq/HTSeq-0.9.1-foss-2016b-Python-2.7.12.eb]: @SethosII The format is wrong here... If you specify the command as a tuple, you're specifying `(<command>, <input to send to stdin>)`. Somehow that's OK to run?

To trigger the issue reported in #5015, try using:

```python
sanity_check_commands = ["htseq-count --help"]

File: easybuild/easyconfigs/h/HTSeq/HTSeq-0.9.1-foss-2016b-Python-2.7.12.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+
+source_urls = [PYPI_SOURCE]
+sources = [SOURCE_TAR_GZ]
+
+checksums = ['af5bba775e3fb45ed4cde64c691ebef36b0bf7a86efd35c884ad0734c27ad485']
+
+dependencies = [
+    ('numpy', '1.10.1', versionsuffix),
```
Comment (boegel) [inline easybuild/easyconfigs/h/HTSeq/HTSeq-0.9.1-foss-2016b-Python-2.7.12.eb]: Is this specific version of `numpy` required?

The `Python-2.7.12-foss-2016b` module already includes `numpy` v1.11.1...

Comment (SethosII) [inline easybuild/easyconfigs/h/HTSeq/HTSeq-0.9.1-foss-2016b-Python-2.7.12.eb]: @boegel Ok, didn't know that. I will remove it.

File: easybuild/easyconfigs/h/HTSeq/HTSeq-0.9.1-foss-2016b-Python-2.7.12.eb (hunk 1 above)
Comment (SethosII) [inline easybuild/easyconfigs/h/HTSeq/HTSeq-0.9.1-foss-2016b-Python-2.7.12.eb]: @boegel I changed it as proposed but the sanity check still completes succesfully and when the command is executed after the module is loaded it still fails?

**Comment by SethosII** [comment]
@boegel I added the requested changes but still got the error with `multiarray` on loading `numpy`

**Comment by boegel** [comment]
@SethosII The `multiarray` import issue you are running into is not a general problem with this easyconfig, it's a system-specific issue, see discussion in #5015

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2027.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/15c4f78d99d1ccf263cfbc1b17955735 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2471.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/56c48cee3be9cc8995a737c11c704300 for a full test report.

**Comment by boegel** [comment]
@SethosII This looks good to merge, are you OK with figuring out the problem you're seeing in #5015?

**Comment by SethosII** [comment]
@boegel Yes, the problem seems to be found, see https://github.com/easybuilders/easybuild-easyconfigs/issues/5015#issuecomment-323058197.

**Comment by boegel** [comment]
Going in, thanks @SethosII!

**Pull request merged**